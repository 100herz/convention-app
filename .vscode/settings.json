{
  "editor.codeActionsOnSave": {
    "source.fixAll": true
  },
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnPaste": true,
  "editor.formatOnSave": true,

  "files.exclude": {
    "**/.git": true,
    "**/.DS_Store": true
  },
  "files.watcherExclude": {
    "**/node_modules/**": true
  },

  // Extensions
  // ----------

  // Jest
  "jest.rootPath": "src/",
  "jest.showCoverageOnLoad": true,
  "jest.enableSnapshotUpdateMessages": false
}
