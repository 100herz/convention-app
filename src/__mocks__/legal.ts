import { Legal } from '@models/legal'

export const legalArray: Legal[] = [
  { id: 1, title: { rendered: 'Impressum' }, content: { rendered: 'Dummy Impressum content' } },
  { id: 2, title: { rendered: 'Datenschutz' }, content: { rendered: 'Dummy Datenschutz content' } },
]
