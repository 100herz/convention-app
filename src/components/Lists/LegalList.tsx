import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
// import { useNavigation } from '@react-navigation/native'
// import { StackNavigationProp } from '@react-navigation/stack'

import Text from '@components/UI/Text'
import Touchable from '@components/UI/Touchable'
// import { RootStackParamList } from '@navigations/HomeNavigator'
import { Legal } from '@models/Legal'
import { colors, defaultStyles, DefaultStyles } from '@styles/theme'

interface Props {
  data: Legal[]
}

const LegalList: React.FC<Props> = ({ data }) => {
  // const navigation = useNavigation<StackNavigationProp<RootStackParamList>>()

  return (
    <View style={styles.listContainer}>
      {data.map(item => (
        <Touchable
          key={item.id}
          // onPress={() => navigation.navigate('ArticlesOverviewScreen', { LegalId: item.id, LegalName: item.name })}
          onPress={() => {}} // eslint-disable-line @typescript-eslint/no-empty-function
          testID="link"
        >
          <View style={styles.listItemContainer}>
            <Text>{item.name}</Text>
            <Ionicons name="ios-arrow-round-forward" size={20} color={colors.primaryColor} />
          </View>
        </Touchable>
      ))}
    </View>
  )
}

const styles = StyleSheet.create<DefaultStyles>({
  ...defaultStyles,
})

export default LegalList
