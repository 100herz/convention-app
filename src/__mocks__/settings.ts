import { Setting } from '@models/settings'

export const settingItem: Setting = { id: 1, name: 'Push Mitteilungen' }

export const settingArray: Setting[] = [{ id: 1, name: 'Push Mitteilungen' }]
