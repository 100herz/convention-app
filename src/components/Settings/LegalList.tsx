import React from 'react'
import { Linking, StyleSheet, View } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import { useNavigation } from '@react-navigation/native'
import { StackNavigationProp } from '@react-navigation/stack'
import HTML from 'react-native-render-html'

import Touchable from '@components/UI/Touchable'
import Text from '@components/UI/Text'
import { SettingsStackParamList } from '@navigations/SettingsNavigator'
import { Legal } from '@models/settings'
import { legalAddition } from '@data/settings'
import { colors, defaultStyles, DefaultStyles } from '@styles/theme'

interface Props {
  data: Legal[]
}

const LegalList: React.FC<Props> = ({ data }) => {
  const navigation = useNavigation<StackNavigationProp<SettingsStackParamList>>()

  const linkIcon = () => <Ionicons name="ios-arrow-round-forward" size={20} color={colors.primaryColor} />

  return (
    <View style={styles.listContainer}>
      {data.map(item => (
        <Touchable
          key={item.id}
          onPress={() => navigation.navigate('LegalScreen', { pageId: item.id, screenTitle: item.title.rendered })}
          testID={`link-${item.id}`}
        >
          <View style={styles.listItemContainer}>
            <HTML baseFontStyle={styles.text} html={item.title.rendered} />
            {linkIcon()}
          </View>
        </Touchable>
      ))}

      <Touchable onPress={() => Linking.openURL(legalAddition.href)} testID="link-mail">
        <View style={styles.listItemContainer}>
          <Text>{legalAddition.name}</Text>
          {linkIcon()}
        </View>
      </Touchable>
    </View>
  )
}

const styles = StyleSheet.create<DefaultStyles>({
  ...defaultStyles,
})

export default LegalList
