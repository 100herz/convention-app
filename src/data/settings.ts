import { Legal, LegalAddition, Setting } from '@models/settings'

export const legalScreens: Legal[] = [
  {
    id: 79,
    title: { rendered: 'Impressum' },
    content: { rendered: '' },
  },
  {
    id: 1105,
    title: { rendered: 'Datenschutz' },
    content: { rendered: '' },
  },
]

export const legalAddition: LegalAddition = {
  id: 1,
  name: 'Anfrage stellen',
  href: 'mailto:anfrage@convention-net.de',
}

export const settingArray: Setting[] = [
  {
    id: 1,
    name: 'Push Mitteilungen',
  },
]
