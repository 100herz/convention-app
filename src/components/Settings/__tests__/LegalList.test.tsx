import React from 'react'
import { Linking } from 'react-native'
import { act, fireEvent, render } from '@testing-library/react-native'

import LegalList from '../LegalList'
import { legalAddition } from '@data/settings'
import { legalArray } from '@__mocks__/legal'

export const mockedNavigate = jest.fn()
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native')
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useNavigationParam: jest.fn(jest.requireActual('@react-navigation/native').useNavigationParam),
  }
})

describe('<LegalList />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<LegalList data={legalArray} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should call the navigate function once', () => {
    const { getByTestId } = render(<LegalList data={legalArray} />)

    act(() => {
      fireEvent.press(getByTestId('link-1'))
    })

    expect(mockedNavigate).toHaveBeenCalledTimes(1)
  })

  it('should call the mailto link', () => {
    const spy = jest.spyOn(Linking, 'openURL')
    const { getByTestId } = render(<LegalList data={legalArray} />)

    act(() => {
      fireEvent.press(getByTestId('link-mail'))
    })

    expect(spy).toBeCalledWith(legalAddition.href)
    spy.mockReset()
    spy.mockRestore()
  })
})
