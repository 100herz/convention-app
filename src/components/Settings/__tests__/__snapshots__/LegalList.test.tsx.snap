// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegalList /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginVertical": 10,
      "paddingHorizontal": 25,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="link-1"
  >
    <View
      style={
        Object {
          "borderBottomColor": "#dfe3e8",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontFamily": "sans",
          }
        }
      >
        Impressum
      </Text>
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="link-2"
  >
    <View
      style={
        Object {
          "borderBottomColor": "#dfe3e8",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontFamily": "sans",
          }
        }
      >
        Datenschutz
      </Text>
      <Text />
    </View>
  </View>
</View>
`;
